test/actions/BackendAction-test.js

```diff
@@ -393,15 +393,56 @@ describe('BackendAction', () => {
     })
 
     it('should work', () => {
+      appSettings.loadAttachedExtensions = () => { return {testExtension: {path: '..'}} }
       backendAction._startSubProcess = () => {}
       try {
+        backendAction.writeExtensionConfigs = () => {}
         backendAction.run({})
       } catch (err) {
         assert.ifError(err)
       }
     })
 
+    it('should write the extension configs for the app', done => {
+      backendAction.appSettings.EXTENSIONS_FOLDER = 'extensions'
+      const extensionConfigPath = path.join(
+        backendAction.appSettings.getApplicationFolder(), AppSettings.EXTENSIONS_FOLDER,
+        'test-extension',
+        'extension-config.json')
+
+      fsEx.createFileSync(extensionConfigPath)
+
+      let mockConf = {someKey: 'someVal'}
+      fsEx.writeJson(
+        extensionConfigPath,
+        mockConf,
+        (err) => {
```

Comment:
As discussed this test could probably be done using async/await to be a bit more readable.